Sources/SourceKit/SourceKitServer.swift

```diff
@@ -364,7 +368,23 @@ extension SourceKitServer {
   }
 
   func codeAction(_ req: Request<CodeActionRequest>, workspace: Workspace) {
-    toolchainTextDocumentRequest(req, workspace: workspace, fallback: nil)
+    toolchainTextDocumentRequest(req, workspace: workspace, resultHandler: { result in
+      switch result {
+      case .success(let reply):
+        return .success(req.params.injectMetadata(atResponse: reply))
+      default:
+        return result
+      }
+    }, fallback: nil)
+  }
+
+  func executeCommand(_ req: Request<ExecuteCommandRequest>, workspace: Workspace) {
+    guard let url = req.params.textDocument?.url else {
```

Comment:
I wonder if we should strip the metadata off before sending it to clang.  What do you think?

Reply:
Yeah, since at this point the correct server can be determined we can strip it for Swift as well. Should make things more predictable 👍